mpas_analysis/test/test_climatology.py

```diff
@@ -0,0 +1,228 @@
+                                                    gridFileName,
+                                                    latVarName='lat',
+                                                    lonVarName='lon')
+
+        mappingFileName = '{}/map_obs_{}_1.0x1.0degree_to_0.5x0.5degree_' \
+                          'bilinear.nc'.format(self.test_dir, fieldName)
+        assert os.path.exists(mappingFileName)
+
+        mappingFileName = '{}/mapping.nc'.format(self.test_dir)
+        config.set('oceanObservations', 'sstClimatologyMappingFile',
+                   mappingFileName)
+        climatology.write_observations_mapping_file(config,
+                                                    componentName,
+                                                    fieldName,
+                                                    gridFileName,
+                                                    latVarName='lat',
+                                                    lonVarName='lon')
+        assert os.path.exists(mappingFileName)
+
+    def test_get_mpas_climatology_file_names(self):
+        config = self.setup_config()
+        fieldName = 'sst'
+        monthNames = 'JFM'
+        (climatologyFileName, regriddedFileName) = \
+            climatology.get_mpas_climatology_file_names(config, fieldName,
+                                                        monthNames)
+        expectedClimatologyFileName = '{}/clim/mpas/sst_QU240_JFM_' \
+                                      'years0002-0002.nc'.format(self.test_dir)
+        self.assertEqual(climatologyFileName, expectedClimatologyFileName)
+
+        expectedRegriddedFileName = '{}/clim/mpas/regrid/sst_QU240_to_' \
+                                    '0.5x0.5degree_JFM_' \
+                                    'years0002-0002.nc'.format(self.test_dir)
+        self.assertEqual(regriddedFileName, expectedRegriddedFileName)
+
+    def test_get_observation_climatology_file_names(self):
+        config = self.setup_config()
+        fieldName = 'sst'
+        monthNames = 'JFM'
+        gridFileName = str(self.datadir.join('obsGrid.nc'))
+        componentName = 'ocean'
+        (climatologyFileName, regriddedFileName) = \
+            climatology.get_observation_climatology_file_names(
+                config, fieldName, monthNames, componentName, gridFileName,
+                latVarName='lat', lonVarName='lon')
+        expectedClimatologyFileName = '{}/clim/obs/sst_1.0x1.0degree_' \
+                                      'JFM.nc'.format(self.test_dir)
+        self.assertEqual(climatologyFileName, expectedClimatologyFileName)
+
+        expectedRegriddedFileName = '{}/clim/obs/regrid/sst_1.0x1.0degree_' \
+                                    'to_0.5x0.5degree_' \
+                                    'JFM.nc'.format(self.test_dir)
+        self.assertEqual(regriddedFileName, expectedRegriddedFileName)
+
+    def open_test_ds(self, config, calendar):
```

Comment:
This is good because it will verify that the time selection will produce the appropriate dataset in terms of length-- great!